nrf_security/src/mbedtls/CMakeLists.txt

```diff
@@ -121,43 +121,43 @@ PREPEND(src_tls_replacement ${NRF_MBEDTLS_ROOT}/src/mbedtls/replacements/
 zephyr_include_directories(${ARM_MBEDTLS_ROOT}/include)
 zephyr_include_directories(${ARM_MBEDTLS_ROOT}/include/mbedtls)
 
-add_library(mbedcrypto_vanilla STATIC "")
-set(ZEPHYR_CURRENT_LIBRARY mbedcrypto_vanilla)
+add_library(${IMAGE}mbedcrypto_vanilla STATIC "")
+set(ZEPHYR_CURRENT_LIBRARY ${IMAGE}mbedcrypto_vanilla)
 zephyr_library_sources(${src_crypto} ${src_crypto_replacement})
 zephyr_library_include_directories(${common_includes})
-zephyr_library_link_libraries(zephyr_interface)
+zephyr_library_link_libraries(${IMAGE}zephyr_interface)
 zephyr_library_compile_definitions_ifdef(CONFIG_MBEDTLS_CIPHER_AES_256_ECB_C MBEDTLS_CIPHER_AES_256_ECB_C)
 zephyr_library_compile_definitions_ifdef(CONFIG_MBEDTLS_CIPHER_AES_256_CBC_C MBEDTLS_CIPHER_AES_256_CBC_C)
 zephyr_library_compile_definitions_ifdef(CONFIG_MBEDTLS_CIPHER_AES_256_CTR_C MBEDTLS_CIPHER_AES_256_CTR_C)
 zephyr_library_compile_definitions_ifdef(CONFIG_MBEDTLS_CIPHER_AES_256_CCM_C MBEDTLS_CIPHER_AES_256_CCM_C)
 if(TARGET mbedcrypto_cc310)
-  target_link_libraries(mbedcrypto_vanilla PRIVATE mbedcrypto_cc310)
+  target_link_libraries(${IMAGE}mbedcrypto_vanilla PRIVATE ${IMAGE}mbedcrypto_cc310)
```

Comment:
Could you use `zephyr_link_libraries(${IMAGE}mbedcrypto_cc310)` here?

Reply:
no, cause that would cause it to be `interface` linking on zephyr_interface target and thus propagated to all targets.
mbedcrypto_cc310 is an imported target that should just be needed by vanilla target here.